In [ ]:
import numpy as np
import pandas as pd
from tools.greedy import Greedy
import os

In [ ]:
archive = "WorkSpace 1000_50_5"
k=5
m=50

In [ ]:
solucion=[13, 20, 23, 29, 49]

In [ ]:
folder_distances = "./data/distances/demand/"
route_distances = os.path.join(folder_distances, f"{archive}.csv")

df_distances_demand = pd.read_csv(route_distances, index_col=0)

In [ ]:
asignacion = df_distances_demand.iloc[:, solucion].idxmin(axis=1)

In [ ]:
print(df_distances_demand.iloc[:, solucion])

             13          20           23          29           49
0    234.000000  556.138472  1063.989662  611.778555   956.694831
1    270.149958  579.276273  1090.668144  617.234153   993.362975
2    279.685538  634.137209  1135.222445  698.645117   987.008105
3    211.123187  572.060312  1068.951355  656.883551   922.366522
4    302.592135  628.012739  1137.453296  666.532070  1021.376033
..          ...         ...          ...         ...          ...
995  164.073154  484.406854   988.907478  564.925659   887.840639
996  305.181913  578.662250  1095.230569  588.510833  1028.572312
997  244.560831  498.036143  1014.282012  524.843786   962.050414
998  229.706334  541.059146  1050.583171  593.177882   953.471552
999  346.343760  568.031689  1087.538505  541.705640  1062.753499

[1000 rows x 5 columns]


In [ ]:
df=df_distances_demand.iloc[:, solucion].round(0)

In [ ]:
distancia_minima = df.min(axis=1)

In [ ]:
conteo_de_minimos = df.eq(distancia_minima, axis=0).sum(axis=1)
hay_empate = conteo_de_minimos > 1

In [ ]:
columna_minima = df.idxmin(axis=1)

In [ ]:
df_resultados = pd.DataFrame({
    'columna_mas_cercana': columna_minima,
    'distancia_minima': distancia_minima,
    'hay_empate_en_minimo': hay_empate
})

In [ ]:
print(df_resultados[df_resultados['hay_empate_en_minimo'] == True])

    columna_mas_cercana  distancia_minima  hay_empate_en_minimo
435                  13             317.0                  True
827                  13             218.0                  True


# Búsqueda de a cuáles se han asignado los empates:
* El 435 se asigna al 20, pero debería asignarse al 13 (en caso de redondear las distancias claro)
* El 827 también se ha asignado al 20. Se le debería asignar al 13.

In [ ]:
df.iloc[827,:]

13    218.0
20    218.0
23    653.0
29    478.0
49    588.0
Name: 827, dtype: float64

In [ ]:
print(list(asignacion[asignacion=='20'].index))

[55, 56, 57, 58, 59, 60, 61, 62, 63, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 115, 226, 227, 255, 278, 279, 280, 315, 316, 319, 320, 321, 326, 327, 328, 354, 355, 363, 426, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 450, 464, 465, 521, 522, 523, 524, 525, 526, 527, 568, 569, 570, 571, 572, 573, 574, 575, 576, 577, 578, 579, 580, 581, 585, 586, 588, 589, 621, 622, 640, 641, 642, 643, 644, 645, 646, 647, 648, 649, 650, 654, 655, 656, 657, 658, 659, 660, 661, 662, 664, 665, 716, 717, 721, 722, 728, 729, 731, 741, 742, 745, 764, 765, 778, 779, 780, 806, 818, 823, 824, 825, 826, 827, 832, 833, 834, 835, 838, 839, 840, 848, 849, 850, 851, 852, 881, 882, 883, 884, 915, 916, 917, 918, 919, 920, 921, 922, 923, 924, 925, 926, 927, 928, 929, 930, 931, 932, 933, 934, 935, 936, 937, 938, 939, 940, 941, 942, 943, 944, 945, 946, 947, 948, 9

In [ ]:
print(list(asignacion[asignacion=='13'].index))

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 222, 223, 224, 225, 228, 229, 230, 256, 257, 258, 259, 260, 261, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 427, 428, 429, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 506, 507, 508, 509, 510, 511, 512, 513, 514, 515, 516, 517, 518, 519, 520, 667, 668, 672, 673, 675, 676, 680, 681, 682, 683, 684, 685, 686, 713, 714, 715, 718, 719, 723, 724, 725, 726, 727, 746, 747, 748, 749, 750, 751, 752, 769, 770, 781, 782, 783, 784, 785, 799, 800, 801, 802, 803, 804, 805, 807, 808, 809, 810, 811, 812, 819, 820, 821, 822, 828, 829, 830, 831, 855, 856, 857, 858, 859, 860, 861, 862, 863, 864, 868, 869, 870, 871, 873, 874, 875, 876,

In [ ]:
pd.value_counts(asignacion)

C:\Users\User\AppData\Local\Temp\ipykernel_32744\2214086861.py:1: FutureWarning: pandas.value_counts is deprecated and will be removed in a future version. Use pd.Series(obj).value_counts() instead.
  pd.value_counts(asignacion)


20    233
13    228
23    224
29    211
49    104
Name: count, dtype: int64